src/sourmash/sketchcomparison.py

```diff
@@ -0,0 +1,138 @@
+"""
+Sketch Comparison Classes
+"""
+import numpy as np
+from dataclasses import dataclass
+
+from .signature import MinHash
+
+
+@dataclass
+class BaseMinHashComparison:
+    """Class for standard comparison between two MinHashes"""
+    mh1: MinHash
+    mh2: MinHash
+    ignore_abundance: bool = False # optionally ignore abundances
+
+    def check_comparison_compatibility(self):
```

Comment:
I tried using that, but I don't want it to check scaled, since I haven't downsampled yet. Is there a way to have it ignore scaled?

Or I guess I could downsample first. I figured it was faster to check the easy things before downsampling, but rust checks are probably faster still, so 🤷‍♀️

Reply:
Maybe just put in a note here, then. At least at the moment this isn't performance critical code, right?

Reply:
fixed in 4e06116c4d8a4a6bc573935d1ca019ac68e0773f

I noticed `_is_compatible` also checks `seed`, which I wasn't doing, so figured it was worth downsampling first to take advantage of that extra check.